File: c/damage_calc.cpp
#include <stdio.h>
#include "damage_func.cpp"

int main(void)
{
	int attack_formation, damaged_condition, attack_mode, firepower, enemy_armor, now_hp, loopnumber;
	printf(">>陣形選択/n#1:単縦陣\n#2:複縦陣\n#3:輪形陣\n#4:梯形陣\n#5:単横陣\n:");
	scanf("%d", &attack_formation);
	printf("\n>>艦状態\n#1:小破以下\n#2:中破\n#3:大破\n:");
	scanf("%d", &damaged_condition);
	printf("\n>>砲撃か雷撃\n#1:砲撃\n#2:雷撃\n:");
	scanf("%d", &attack_mode);
	printf("\n>>火力値(雷装値):");
	scanf("%d", &firepower);
	printf("\n>>敵の装甲値:");
	scanf("%d", &enemy_armor);
	printf("\n>>現在の敵HP:");
	scanf("%d", &now_hp);
	printf("\n>>繰り返す回数:");
	scanf("%d", &loopnumber);
	
	float correction_before_cap[];
	float basic_attackpower[];
	float attackpower_before_cap[];
	float attackpower_after_cap[];
	float final_attackpower[];
	float vitality[];
	float results_list[];
	int prob_hit, prob_throught_the_armor, prob_percentage_damage, prob_miss, miss_counter = 0;
	
	printf("\n######################################################");
	
	for(i=0; i<loopnumber; i++){
		int engagement_form;
		engagement_form = engagement_form_func();
		#printf("%d", engagement_form);

		correction_before_cap[i] = correction_before_cap_func(engagement_form, attack_formation, damaged_condition, attack_mode);
		#printf("\n>>キャップ前補正:%f", correction_before_cap[i]);

		basic_attackpower[i] = basic_attackpower(firepower);
		#printf("\n>>基本攻撃力:%f", basic_attackpower[i]);

		attackpower_before_cap[i] = attackpower_before_cap_func(basic_attackpower[i], correction_before_cap[i]);
		#printf("\n>>キャップ前攻撃力:%f", attackpower_before_cap[i]);
		
		attackpower_after_cap[i] = attackpower_after_cap_func(attackpower_before_cap[i]);
		#printf("\n>>キャップ後攻撃力:%f", attackpower_after_cap[i]);
		
		final_attackpower[i] = final_attackpower_func(attackpower_after_cap[i]);
		#printf("\n>>最終攻撃力:%f", final_attackpower[i]);
}

File: python/damage_calc.py
import Kancolle_damage_func as kd

attack_formation = input('''>>陣形選択
#1:単縦陣
#2:複縦陣
#3:輪形陣
#4:梯形陣
#5:単横陣
:''')
damaged_condition = input('''\n>>艦状態
#1:小破以下
#2:中破
#3:大破
:''')
attack_mode = input('''\n>>砲撃か雷撃
#1:砲撃
#2:雷撃
:''')

firepower = input("\n>>火力値(雷装値):")
enemy_armor = input("\n>>敵の装甲値:")
now_hp = input("\n>>現在の敵HP:")

loopnumber = input("\n>>繰り返す回数:")

##################################################################

correction_before_cap = []
basic_attackpower = []
attackpower_before_cap = []
attackpower_after_cap = []
final_attackpower = []
vitality = []
results_list = []
prob_hit = 0
prob_through_the_armor = 0
prob_percentage_damage = 0
prob_miss = 0
miss_counter = 0

for i in range(0, int(loopnumber)):
    engagement_form = kd.engagement_form_func()

    correction_before_cap.append(kd.correction_before_cap_func(engagement_form, attack_formation, damaged_condition, attack_mode))
    #print(f"\n>>キャップ前補正:{correction_before_cap[i]}")

    basic_attackpower.append(kd.basic_attackpower_func(firepower))
    #print(f">>基本攻撃力:{basic_attackpower[i]}")

    attackpower_before_cap.append(kd.attackpower_before_cap_func(basic_attackpower[i], correction_before_cap[i]))
    #print(f">>キャップ前攻撃力:{attackpower_before_cap[i]}")

    attackpower_after_cap.append(kd.attackpower_after_cap_func(attackpower_before_cap[i], attack_mode))
    #print(f">>キャップ後攻撃力:{attackpower_after_cap}")

    final_attackpower.append(kd.final_attackpower_func(attackpower_after_cap[i]))
    #print(f">>最終攻撃力:{final_attackpower}")

    vitality.append(kd.vitality_func(enemy_armor))
    #print(f">>敵の防御力:{vitality}")

    damage = kd.damage_func(final_attackpower[i], vitality[i], 1)
    if damage <= 0:
        percentage_damage = kd.percentage_damage_func(int(now_hp))
        if percentage_damage <= 0:
            results_list.append("miss")
            prob_miss += 100
            miss_counter += 1
            #print("\n>>結果:miss")
        else:
            results_list.append(str(percentage_damage) + "(カスダメ)")
            prob_percentage_damage += 100
            prob_hit += 100
            #print("\n>>カスダメ")
            #print(f">>結果:{percentage_damage}")
    else:
        results_list.append(damage)
        prob_through_the_armor += 100
        prob_hit += 100
        #print(f"\n>>結果:{damage}")
    
############################################################################
    
    print(f"\n>>{i+1}回目のキャップ前補正:{correction_before_cap[i]}")
    print(f">>{i+1}回目の基本攻撃力:{basic_attackpower[i]}")
    print(f">>{i+1}回目のキャップ前攻撃力:{attackpower_before_cap[i]}")
    print(f">>{i+1}回目のキャップ後攻撃力:{attackpower_after_cap[i]}")
    print(f">>{i+1}回目の最終攻撃力:{final_attackpower[i]}")
    print(f">>{i+1}回目の敵の防御力:{vitality[i]}")
    print(f">>{i+1}回目の結果:{results_list[i]}")

    
